cmd/posix.go

```diff
@@ -1605,10 +1607,12 @@ func (s *posix) VerifyFile(volume, path string, fileSize int64, algo BitrotAlgor
 
 		h := algo.New()
 		if _, err = io.CopyBuffer(h, file, *bufp); err != nil {
-			return err
+			// Premature failure in reading the object,file is corrupt.
+			return errFileCorrupt
 		}
 		if !bytes.Equal(h.Sum(nil), sum) {
-			return HashMismatchError{hex.EncodeToString(sum), hex.EncodeToString(h.Sum(nil))}
+			return fmt.Errorf("hashes do not match expected %s, got %s",
```

Comment:
I think we should wrap this error with VerifyFileError() so shouldHealObjectOnDisk() can detect if an object needs to be healed or not.

Reply:
Wrapping is already done in defer.